### PR TITLE
feat!: Enable acceleration support, raies MSV or Terraform and AWS provider

### LINKED ISSUE #98: Enable acceleration flag when use TGW
## Is your request related to a new offering from AWS?
Is this functionality available in the [AWS provider for Terraform](https://registry.terraform.io/providers/hashicorp/aws/latest/docs)? 

  - Yes ✅: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/vpn_connection#enable_acceleration

## Is your request related to a problem? Please describe.
To use Global Accelerator for low latency and high throughput, you can enable acceleration on site-to-site VPN connections.

## Describe the solution you'd like.
An option enable_acceleration would be great that, maybe in combination with a check if connect_to_transit_gateway is set as this is only usable with tgw.

## Describe alternatives you've considered.
Can't see any.

## Additional context

I have seen same issues and feat requests for adding this flag. Why is the requests was ingnored? It's very useful flag

### PR DESCRIPTION
**Pull request by james-martinez**
## Description

Added bool var for acceleration support. 
Added terraform parameter in each vpn resource to reference the var.

An accelerated Site-to-Site VPN connection (accelerated VPN connection) uses AWS Global Accelerator to route traffic from your on-premises network to an AWS edge location that is closest to your customer gateway device. AWS Global Accelerator optimizes the network path, using the congestion-free AWS global network to route traffic to the endpoint that provides the best application performance (for more information, see [AWS Global Accelerator](https://aws.amazon.com/global-accelerator/)).
## Motivation and Context

https://docs.aws.amazon.com/vpn/latest/s2svpn/accelerated-vpn.html

Resolves #98 
## Breaking Changes

No breaking changes

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

**Comment by antonbabenko** [comment]
Please fix the failing CI runs (and docs).

**Comment by james-martinez** [comment]
Can someone test the example? I don't have an aws account to deploy the example.

Thanks

**Comment by dmildh-absci** [comment]
I have also confirmed the`vpn_connection_enable_acceleration   = true` option is working if someone can merge this.

File: variables.tf [hunk 1]
```diff
@@ -53,6 +53,12 @@ variable "vpn_connection_static_routes_only" {
   default     = false
 }
 
+variable "vpn_connection_enable_acceleration" {
+  description = "Indicate whether to enable acceleration for the VPN connection. Supports only EC2 Transit Gateway."
+  type        = bool
+  default     = false
```
Comment (bryantbiggs) [inline variables.tf:59]:
```suggestion
  default     = null
```

**Comment by shmick** [comment]
Is there anything else needed for this to be reviewed and merged? Having acceleration support would be great.

**Review by bryantbiggs — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 4.0.0 :tada:

**Pull request merged**